flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/AvroToRowDataConverters.java

```diff
@@ -0,0 +1,406 @@
+        return avroObject -> {
+            if (avroObject == null) {
+                return null;
+            }
+            return converter.convert(avroObject);
+        };
+    }
+
+    /** Creates a runtime converter which assuming input object is not null. */
+    private static AvroToRowDataConverter createConverter(LogicalType type) {
+        switch (type.getTypeRoot()) {
+            case NULL:
+                return avroObject -> null;
+            case TINYINT:
+                return avroObject -> ((Integer) avroObject).byteValue();
+            case SMALLINT:
+                return avroObject -> ((Integer) avroObject).shortValue();
+            case BOOLEAN: // boolean
+            case INTEGER: // int
+            case INTERVAL_YEAR_MONTH: // long
+            case BIGINT: // long
+            case INTERVAL_DAY_TIME: // long
+            case FLOAT: // float
+            case DOUBLE: // double
+                return avroObject -> avroObject;
+            case DATE:
+                return AvroToRowDataConverters::convertToDate;
+            case TIME_WITHOUT_TIME_ZONE:
+                int timePrecision = ((TimeType) type).getPrecision();
+                if (timePrecision <= millisPrecision) {
+                    return avroObject -> convertToMillisTime(avroObject, millisPrecision);
+                } else if (timePrecision <= microsPrecision) {
+                    return avroObject -> convertToMicrosTime(avroObject, microsPrecision);
+                } else {
+                    throw new UnsupportedOperationException(
+                            String.format(
+                                    "TIME type with Precision > %d is not supported!",
+                                    microsPrecision));
+                }
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                int localTsPrecision = ((LocalZonedTimestampType) type).getPrecision();
+                return createTimeDatetimeConvertor(localTsPrecision);
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                int tsPrecision = ((TimestampType) type).getPrecision();
+                return createTimeDatetimeConvertor(tsPrecision);
+            case CHAR:
+            case VARCHAR:
+                return avroObject -> StringData.fromString(avroObject.toString());
+            case BINARY:
+            case VARBINARY:
+                return AvroToRowDataConverters::convertToBytes;
+            case DECIMAL:
+                return createDecimalConverter((DecimalType) type);
+            case ARRAY:
+                return createArrayConverter((ArrayType) type);
+            case ROW:
+                return createRowConverter((RowType) type);
+            case MAP:
+            case MULTISET:
+                return createMapConverter(type);
+            case RAW:
+            default:
+                throw new UnsupportedOperationException("Unsupported type: " + type);
```

Comment:
This is a great example of how we should store relevant context in error message while throwing exceptions. By the time this gets propagated up the method stack and wrapped in a generic connector exception, all context about the cause would've been lost.
Please replace 
```
throw new UnsupportedOperationException("Unsupported type: " + type);
```
with 
```
throw new UnsupportedOperationException("Cannot create avro to Flink table's LogicalType for LogicalType: " + type.asSummaryString());
```
Then we can go on to explain more about this particular error in documentation (readme or faq etc) as to what this error means (BQ table to avro to row data) and how to fix it.

Reply:
Fixed.